ui/pages/settings/networks-tab/networks-form/networks-form.js

```diff
@@ -413,6 +463,22 @@ const NetworksForm = ({
         }
       }
 
+      if (
+        Object.values(orderedNetworksList).some(
+          (network) =>
+            getDisplayChainId(chainArg) ===
+              parseInt(network.networkId, 16).toString(10) &&
+            rpcUrl === network.networkRpcUrl,
+        ) &&
+        addNewNetwork
```

Comment:
Performance Nit: If `addNewNetwork` is the first condition in the `if` we can avoid a `orderedNetworksList` search.